src/dialog.jsx

```diff
@@ -379,27 +375,30 @@ let Dialog = React.createClass({
     }
   },
 
-  _onShow() {
-    if (this.props.onShow) this.props.onShow();
+  _show() {
+    this.refs.dialogOverlay.preventScrolling();
+    this.setState({ open: true });
+  },
+
+  _dismiss() {
+    CssEvent.onTransitionEnd(ReactDOM.findDOMNode(this), () => {
+      this.refs.dialogOverlay.allowScrolling();
+    });
+
+    this.setState({ open: false });
   },
 
-  _onDismiss() {
-    if (this.props.onDismiss) this.props.onDismiss();
+  _requestClose() {
+    if (this.props.onRequestClose) this.props.onRequestClose();
```

Comment:
I also think that we should pass some sort of reason for the close request since pressing one of the default action buttons will also trigger `_requestClose` right now. When attempting to create a `modal` Dialog there is no way of knowing if the Dialog is trying to close by means of a button or by touching/clicking outside of the Dialog.